crypto/default/src/test/java/org/keycloak/crypto/def/test/DefaultCryptoRSAVerifierTest.java

```diff
@@ -1,10 +1,19 @@
 package org.keycloak.crypto.def.test;
 
+import org.junit.Assume;
+import org.junit.Before;
 import org.keycloak.RSAVerifierTest;
+import org.keycloak.common.util.Environment;
 
 /**
  * Test with bouncycastle security provider
  * 
  */
 public class DefaultCryptoRSAVerifierTest extends RSAVerifierTest {
+
+    @Before
+    public void before() {
+        // Run this test just if java is not in FIPS mode
+        Assume.assumeFalse("Java is in FIPS mode. Skipping the test.", Environment.isJavaInFipsMode());
```

Comment:
Would be good to detach "test mode" from "server mode". That can be done in a follow-up PR for sure.

Basically the tests should be executed explicitly in FIPS mode rather than just rely on the "auto-detect" stuff working properly.

Reply:
+1

Right now it is already possible to use the system property `com.redhat.fips` to explicitly enforce if tests should be triggered in the FIPS mode (with BCFIPS) or in the non-fips mode. The auto-detect is used just if this property is not provided. But it will need to be addressed more properly though.